genie-web/src/main/java/com/netflix/genie/web/tasks/job/JobCompletionService.java

```diff
@@ -376,79 +376,108 @@ private JobStatus updateFinalStatusForJob(final String id) throws GenieException
-                        + JobConstants.DEPENDENCY_FILE_PATH_PREFIX
-                );
+                final Optional<String> clusterId = this.jobSearchService.getJobCluster(jobId).getId();
+                if (clusterId.isPresent()) {
+                    dependencyDirectories.add(
+                        new File(
+                            jobDir,
+                            JobConstants.GENIE_PATH_VAR
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + JobConstants.CLUSTER_PATH_VAR
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + clusterId.get()
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + JobConstants.DEPENDENCY_FILE_PATH_PREFIX
+                        )
+                    );
+                }
+            } catch (GenieException e) {
+                log.error(
+                    "Error collecting cluster dependency for job: {} due to error {}",
+                    jobId,
+                    e);
+                incrementErrorCounter("DELETE_CLUSTER_DEPENDENCIES_FAILURE", e);
+            }
 
-                deleteDependenciesDirectory(clusterDependencyDir);
+            // Collect command dependencies
+            try {
+                final Optional<String> commandId = this.jobSearchService.getJobCommand(jobId).getId();
+                if (commandId.isPresent()) {
+                    dependencyDirectories.add(
+                        new File(
+                            jobDir,
+                            JobConstants.GENIE_PATH_VAR
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + JobConstants.COMMAND_PATH_VAR
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + commandId.get()
+                                + JobConstants.FILE_PATH_DELIMITER
+                                + JobConstants.DEPENDENCY_FILE_PATH_PREFIX
+                        )
+                    );
+                }
+            } catch (GenieException e) {
+                log.error(
+                    "Error collecting command dependency for job: {} due to error {}",
+                    jobId,
+                    e);
+                incrementErrorCounter("DELETE_COMMAND_DEPENDENCIES_FAILURE", e);
+            }
 
-            } catch (Exception e) {
-                log.error("Could not delete job dependencies after completion for job: {} due to error {}",
-                    jobId, e);
-                incrementErrorCounter("DELETE_CLUSTER_DEPENDENCIES_FAILURE", e);
+            // Delete all dependencies
+            for (File dependencyDirectory : dependencyDirectories) {
+                try {
+                    deleteDependenciesDirectory(dependencyDirectory);
+                } catch (IOException e) {
+                    log.error(
```

Comment:
So you have a counter above for getting the files but you don't have an error counter if we actually get an error here?

Reply:
The counter is inside `deleteDependenciesDirectory()`, however since this is now the only place that this is called from, I'm going to inline it so all counters and logic are in one self contained method